### PR TITLE
[CSL-1453] Drop wreq from the stack

### PR DESCRIPTION
**Pull request by adinapoli-iohk**
It turned out to be quite easy to do, as we were using `wreq` in just one module. I have then run my `dependencies.sh` script to indeed prove `wreq` is now gone:

```
☁  iohk  cat deps_master.txt | grep wreq | wc -l
       0
```

### CODE REVIEW

File: infra/cardano-sl-infra.cabal [hunk 1]
```diff
@@ -126,6 +126,8 @@ library
                       , formatting
                       , generic-arbitrary
                       , hashable
+                      , http-client
```
Comment (adinapoli-iohk) [inline infra/cardano-sl-infra.cabal]: We depend from `http-client` and `http-client-tls` anyway, so this doesn't add any extra dep, in fact.

File: infra/Pos/Reporting/Methods.hs [hunk 2]
```diff
@@ -232,17 +238,25 @@ sendReport logFiles rawLogs reportType appName reportServerUri = do
...
         let pathsPart = map partFile' existingFiles
         let payloadPart =
-                partLBS "payload"
+                Form.partLBS "payload"
                 (encode $ reportInfo curTime $ existingFiles ++ memlogFiles)
-        e <- try $ liftIO $ post (reportServerUri <//> "report") $
-             payloadPart : (memlogPart ++ pathsPart)
+        -- If performance will ever be a concern, moving to a global manager
+        -- should help a lot.
+        reportManager <- liftIO (newManager tlsManagerSettings)
```
Comment (adinapoli-iohk) [inline infra/Pos/Reporting/Methods.hs]: Note how this is not any worse than what `wreq` does, which is creating a new `Manager` everytime, unless there is one passed externally.

File: infra/Pos/Reporting/Methods.hs [hunk 3]
```diff
@@ -232,17 +238,25 @@ sendReport logFiles rawLogs reportType appName reportServerUri = do
     withSystemTempFile "main.log" $ \tempFp tempHandle -> liftIO $ do
         for_ rawLogs $ TIO.hPutStrLn tempHandle
         hClose tempHandle
+        rq0 <- liftIO (parseUrlThrow $ reportServerUri <//> "report")
```
Comment (gromakovsky) [inline infra/Pos/Reporting/Methods.hs]: Hm, it seems all these `liftIO`s inside `do` are redundant because there is `liftIO` outside (3 lines above).

Comment (adinapoli-iohk) [inline infra/Pos/Reporting/Methods.hs]: Ok, well spotted, you are right!

**Pull request merged**